src/main/java/org/killbill/billing/plugin/adyen/client/payment/service/AdyenPaymentServiceProviderPort.java

```diff
@@ -145,6 +151,68 @@ else if (paymentServiceProviderResult == PaymentServiceProviderResult.IDENTIFY_S
         return purchaseResult;
     }
 
+    public PurchaseResult authoriseKlarnaPayment(final String merchantAccount,
+                                                 final PaymentData paymentData,
+                                                 final UserData userData) {
+        PaymentsRequest klarnaRequest = adyenRequestFactory.createKlarnaPayment(merchantAccount, paymentData, userData);
+        PurchaseResult purchaseResult = null;
+        try {
+            PaymentsResponse response = adyenCheckoutApiClient.createPayment(klarnaRequest);
+            purchaseResult = extractKlarnaResponse(response);
+        }
+        catch (ApiException ex) {
+            new PurchaseResult(PaymentServiceProviderResult.valueOf("error"),
```

Comment:
if we are not doing anything extra then what is the use of adding different exceptions.